nodes/common/pulp_node/conduit.py

```diff
@@ -13,77 +16,99 @@ def get_units(self, repo_id):
         :return: unit iterator
         :rtype: UnitsIterator
         """
-        units = {}
-        types = {}
+        unit_ids = {}
+        associations = {}
         collection = RepoContentUnit.get_collection()
-        for unit in collection.find({'repo_id': repo_id}):
-            unit_id = unit['unit_id']
-            type_id = unit['unit_type_id']
-            units[unit_id] = unit
-            unit_list = types.setdefault(type_id, [])
-            unit_list.append(unit['unit_id'])
-        return UnitsIterator(units, types)
-
-
-class Typedef(object):
-
-    def __init__(self):
-        self.cached = {}
+        for association in collection.find({'repo_id': repo_id}):
```

Comment:
Can this loop be redone using mongoengine based queries? We shouldn't continue to use direct pymongo based code anymore since the collection being accessed is converted to mongoengine. I bet this whole thing would become perhaps one or two lines.